internal/mage/engine.go

```diff
@@ -114,6 +117,16 @@ func (t Engine) Publish(ctx context.Context, version string) error {
 	time.Sleep(3 * time.Second) // allow buildkit logs to flush, to minimize potential confusion with interleaving
 	fmt.Println("PUBLISHED IMAGE REF:", digest)
 
+	// gpu is experimental, not fatal if publish fails
+	gpuDigest, err := c.Container().Publish(ctx, gpuRef, dagger.ContainerPublishOpts{
+		PlatformVariants: util.DevEngineContainerWithGPUSupport(c, publishedGPUEngineArches, version),
+	})
+	if err == nil {
+		fmt.Println("PUBLISHED GPU IMAGE REF:", gpuDigest)
+	} else {
+		fmt.Println("GPU IMAGE PUBLISH FAILED: ", err.Error())
+	}
+
```

Comment:
While the GPU support is experimental, I am pretty sure that we want to fail if the Engine image publish fails. If someone wants to try out the experimental GPU support, there should be an image to do so. It's OK if some aspect of this image doesn't work, but not having it is not an option in my mind.

Before I go ahead and change this, did I miss something @matiasinsaurralde?

Reply:
This was added by @sipsma, he mentioned:
```
Don't fail the engine publish step if gpu image publishing fails (we just print a log saying it failed)
This is so any unexpected problems are less likely to impact our production image publish pipeline
```

Reply:
OK, that helps.

We definitely want to fail publishing if the GPU image publish fails. Since the Engine image gets published first, before the CLI, it's OK if this step fails. We can fix, re-tag, force push the tag & re-run the publishing. Once the CLI gets published, overwriting those artefacts is a lot trickier, and we should avoid it.

I will make this change next so that I can capture the reasoning in the commit message. We can use that as a starting point for follow-up conversations. cc @sipsma

Reply:
Just pushed the commit that addresses this:https://github.com/dagger/dagger/pull/5605/commits/82ad7c02fc1a6d40679a8c3f835a4ba793892db2

Not resolving the conversation so that it's easier to spot in the future.